### PR TITLE
Use commit information for badge if available

### PR DESCRIPTION
**Pull request by vermiculus**
With #6, we now have commit information available.  When this information is available (and it's a SHA), use it for the badge instead of the package version (e.g. `melpa-06fe4b` instead of `melpa-20170514.1323`).

### CODE REVIEW

File: package-build.el [hunk 1]
```diff
@@ -1287,11 +1287,13 @@ ARCHIVE-ENTRY is destructively modified."
             (when commit
               (package-build-add-to-archive archive-entry :commit commit))
             (package-build--dump archive-entry
-                                 (package-build--entry-file-name archive-entry)))
-          (when package-build-write-melpa-badge-images
-            (package-build--write-melpa-badge-image
-             file-name
-             version package-build-archive-dir))
+                                 (package-build--entry-file-name archive-entry))
+            (when package-build-write-melpa-badge-images
```
Comment (vermiculus) [inline package-build.el:1291]: This `when` form needed to move inside the `let` so we'd have access to `archive-entry`.

File: package-build.el [hunk 2]
```diff
@@ -1613,6 +1615,13 @@ If FILE-NAME is not specified, the default archive-contents file is used."
 ;; separate package.  Note also that it would be straightforward to
 ;; generate the SVG ourselves, which would save the network overhead.
 
+(defun package-build--melpa-badge-image-version (config archive-entry)
+  "For CONFIG, determine a version label for ARCHIVE-ENTRY."
+  (when (memq (plist-get config :fetcher) '(github gitlab))
+    (let ((commit (alist-get :commit (elt (cdr archive-entry) 4))))
+      (when (and commit (stringp commit) (= 40 (length commit)))
```
Comment (vermiculus) [inline package-build.el:1622]: As I understand it, the MELPA server is running on an old Emacs (i.e., an emacs without `when-let`).  If theres another macro in-scope that I could use, let me know.

**Comment by purcell** [comment]
I'm -1 on this, sorry. It's much clearer for users if they can see when the last package was made, ie. the date. The page the badge links to will show the commit, so I think we should keep things as they are.

**Comment by vermiculus** [comment]
Alright, but if the intention is to show when the package was made, why do we not format it as a date (i.e., `format-time-string`)? Is there value in showing when it was packaged versus when it was last updated? (Between *those* two, personally I'm for the former (i.e. existing) behavior.)

**Comment by purcell** [comment]
The time is the version of the last-built package, so it's the standard identifier that people can use to see at a glance if they've got the latest version. Or am I not quite understanding your question?

**Comment by vermiculus** [comment]
There are two possibilities:
* version is `(current-time)` at the point we clone the repository
* version is derived from commit information
Pretty sure it's the former for no better reason than it would be more straightforward to implement.

Either way, are we expecting people to be using the badges to compare against what version they have? I'm not sure that's a realistic use-case; I'd imagine most people are using a package manager inside emacs to determine that (where the badge doesn't come into play at all).

**Comment by purcell** [comment]
> Either way, are we expecting people to be using the badges to compare against what version they have? 

Yes, I and others do that all the time. However we define the package version, that's what should show up in the badge. There's plenty of discussion on that topic in https://github.com/melpa/melpa/issues/2955 and other issues.

**Pull request closed without merging**